### PR TITLE
The type () is replaced with Unit

### PR DESCRIPTION
**Pull request by michel-steuwer**
The type () is replaced with Unit (as Scala 3 seems to no longer accept () as a type)

### CODE REVIEW

File: src/main/scala/rise/eqsat/Analysis.scala [hunk 1]
```diff
@@ -294,13 +294,13 @@ object NoAnalysis extends SemiLatticeAnalysis with TypeAnalysis {
   override def requiredTypeAnalyses(): Set[TypeAnalysis] = Set()
 
   override def make(egraph: EGraph, enode: ENode, t: TypeId,
-                    analysisOf: EClassId => ()): () = ()
+                    analysisOf: EClassId => Unit): Unit = ()
```
Comment (Bastacyclop) [inline src/main/scala/rise/eqsat/Analysis.scala:297]:
```scala
object NoAnalysis extends SemiLatticeAnalysis with TypeAnalysis {
  type Data = ()
  type NatData = ()
  type TypeData = ()
```
Should these become `Unit` as well??

**Review by Bastacyclop — APPROVED** [review]

File: src/main/scala/rise/eqsat/Analysis.scala (hunk 1 above)
Comment (michel-steuwer) [reply src/main/scala/rise/eqsat/Analysis.scala:297]: Yes, I guess so. Everywhere where in the type `()` is used to mean Unit.
It's hard to search semantically like this in the codebase.

Could you make this update in the PR?

**Review by Bastacyclop — APPROVED** [review]

File: src/main/scala/rise/eqsat/Analysis.scala (hunk 1 above)
Comment (Bastacyclop) [reply src/main/scala/rise/eqsat/Analysis.scala:297]: @michel-steuwer should be good to merge now

**Comment by Bastacyclop** [comment]
The CI seems to randomly fail due to missed nat simplifications :/

File: src/main/scala/rise/eqsat/Analysis.scala (hunk 1 above)
Comment (michel-steuwer) [reply src/main/scala/rise/eqsat/Analysis.scala:297]: Great. Thanks!

**Pull request merged**